integrations/opensearch/src/haystack_integrations/components/retrievers/open_search_hybrid_retriever.py

```diff
@@ -0,0 +1,444 @@
+        device: Optional[ComponentDevice] = None,
+        normalize_embeddings: bool = False,
+        model_kwargs: Optional[Dict[str, Any]] = None,
+        tokenizer_kwargs: Optional[Dict[str, Any]] = None,
+        config_kwargs: Optional[Dict[str, Any]] = None,
+        encode_kwargs: Optional[Dict[str, Any]] = None,
+        backend: Literal["torch", "onnx", "openvino"] = "torch",
+        # OpenSearchBM25Retriever
+        filters_bm25: Optional[Dict[str, Any]] = None,
+        fuzziness: Union[int, str] = "AUTO",
+        top_k_bm25: int = 10,
+        scale_score: bool = False,
+        all_terms_must_match: bool = False,
+        filter_policy_bm25: Union[str, FilterPolicy] = FilterPolicy.REPLACE,
+        custom_query_bm25: Optional[Dict[str, Any]] = None,
+        # OpenSearchEmbeddingRetriever
+        filters_embedding: Optional[Dict[str, Any]] = None,
+        top_k_embedding: int = 10,
+        filter_policy_embedding: Union[str, FilterPolicy] = FilterPolicy.REPLACE,
+        custom_query_embedding: Optional[Dict[str, Any]] = None,
+        # DocumentJoiner
+        join_mode: Union[str, JoinMode] = JoinMode.CONCATENATE,
+        weights: Optional[List[float]] = None,
+        top_k: Optional[int] = None,
+        sort_by_score: bool = True,
+        # ChatPromptBuilder
+        template: Optional[List[ChatMessage]] = None,
+        # OpenAIChatGenerator
+        generator_model: str = "gpt-4o-mini",
+        streaming_callback: Optional[StreamingCallbackT] = None,
+        api_base_url: Optional[str] = None,
+        organization: Optional[str] = None,
+        generation_kwargs: Optional[Dict[str, Any]] = None,
+        http_client_kwargs: Optional[Dict[str, Any]] = None,
+        # AnswerBuilder
+        pattern: Optional[str] = None,
+        reference_pattern: Optional[str] = None,
+        # extra kwargs
+        **kwargs,
+    ):
+        """
+        Initialize the OpenSearchHybridRetriever, a super component to retrieve documents from OpenSearch using
+        both embedding-based and keyword-based retrieval methods.
+
+        This super component is tied to a SentenceTransformersTextEmbedder and an OpenAIChatGenerator.
+
+        We don't explicitly define all the init parameters of the components in the constructor, for each
+        of the components, since that would be around 50 parameters. Instead, we define the most important ones
+        and pass the rest as kwargs. This is to keep the constructor clean and easy to read.
+
+        If you need to pass extra parameters to the components, you can do so by passing them as kwargs. It expects
+        a dictionary with the component name as the key and the parameters as the value.
+
+        text_embedder -> SentenceTransformersTextEmbedder
+        bm25_retriever -> OpenSearchBM25Retriever
+        embedding_retriever -> OpenSearchEmbeddingRetriever
+        document_joiner -> DocumentJoiner
+        chat_prompt_builder -> ChatPromptBuilder
```

Comment:
Let's not include a ChatPromptBuilder, OpenAIChatGenerator and AnswerBuilder in the HybridRetriever. The HybridRetriever should only retrieve documents from OpenSearch using both embedding-based and keyword-based retrieval methods. That way it can be re-used more flexibly.